gramps/__main__.py

```diff
@@ -0,0 +1,31 @@
+#! /usr/bin/env python3
+#
+# Gramps - a GTK+/GNOME based genealogy program
+#
+# Copyright (C) 2012       Benny Malengier
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301  USA
+#
+
+"""This is a stub to start Gramps.
+
+It is provided to be able to start gramps as `python3 -m gramps`,
+e.g. in a virtual environment.
+"""
+
+import gramps.grampsapp as app
+
+if __name__ == "__main__":
```

Comment:
Will this ever be something other than "__main__"?

Reply:
I just checked that if it is imported by another script, it is set to `"gramps.__main__."`. Of course, it's a bit silly to assume that this will happen, so one can also just drop it.

Reply:
The `if` statement seems redundant.  Is it necessary?

Reply:
It's to some extent convention, but the motivation is that it does not run the code on importing the module (`import gramps.__main__`).

See e.g. here: https://docs.python.org/3/library/__main__.html

Reply:
The `_main__.py` file isn't run on import anyway, just the `__init__.py`  file.

Reply:
I meant when importing `__main__`. I mean, this `if` statement is just there so the code below is only ever executed when this is run as a script. And to be honest I have never seen a `__main__.py` that does *not* have this `if` statement. Prominent examples:

- https://github.com/pypa/pip/blob/master/src/pip/__main__.py#L25
- https://github.com/python/mypy/blob/master/mypy/__main__.py#L22
- https://github.com/pytest-dev/pytest/blob/master/src/pytest/__main__.py#L4
- https://github.com/pypa/twine/blob/master/twine/__main__.py#L52
- https://github.com/python/cpython/blob/master/Lib/ensurepip/__main__.py#L4

It's also a common pattern to have a `main` function in `__main__.py` *before* the block which can be imported.

Reply:
OK.  Thanks.  I'll merge this now.

Reply:
Thanks! :+1: